### PR TITLE
chore: Improves error handling messages in advanced cluster tpf

### PR DESCRIPTION
**Pull request by EspenAlbert**
## Description

Improves error handling messages in advanced cluster tpf
Error diagnostics principle: 
- `Summary` should be a real sentence, static, and unique string. 
- `Detail` can be more sentences with parameters.

Alo changes:
- Include the attribute name for deprecated fields

Link to any related issue(s): CLOUDP-288298

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [ ] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals I have added appropriate changelog entries.
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

**Comment by EspenAlbert** [comment]
Only `TestAccClusterAdvancedCluster_withTags` failing (cloud dev 3h timeout)

File: internal/service/advancedclustertpf/data_source.go [hunk 1]
```diff
@@ -14,6 +14,12 @@ import (
 var _ datasource.DataSource = &ds{}
 var _ datasource.DataSourceWithConfigure = &ds{}
 
+const (
+	errorReadDatasource                 = "Error reading  advanced cluster datasource"
+	errorReadDatasourceAsymmetric       = "Error reading  advanced cluster datasource"
```
Comment (andreaangiolillo) [inline internal/service/advancedclustertpf/data_source.go]: these errors seem the same. Why do you need both?

File: internal/service/advancedclustertpf/data_source.go [hunk 2]
```diff
@@ -14,6 +14,12 @@ import (
...
+	errorReadDatasourceAsymmetricDetail = "Cluster name %s. Please add `use_replication_spec_per_shard = true` to your data source configuration to enable asymmetric shard support. Refer to documentation for more details."
```
Comment (andreaangiolillo) [inline internal/service/advancedclustertpf/data_source.go]: Could you provide a link to the doc as part of the error message? We do this in the CLI. You should ask the doc team for a dochub link if the documentation is in the MDB website

File: internal/service/advancedclustertpf/data_source.go [hunk 3]
```diff
@@ -14,6 +14,12 @@ import (
 var _ datasource.DataSource = &ds{}
 var _ datasource.DataSourceWithConfigure = &ds{}
 
+const (
```
Comment (andreaangiolillo) [inline internal/service/advancedclustertpf/data_source.go:17]: Are these principles documented somewhere in the repo?

> Error diagnostics principle:
Summary should be a real sentence, static, and unique string.
Detail can be more sentences with parameters.
Include the attribute name for deprecated fields

**Review by andreaangiolillo — APPROVED** [review]
LGTM

File: internal/service/advancedclustertpf/data_source.go (hunk 3 above)
Comment (EspenAlbert) [reply internal/service/advancedclustertpf/data_source.go:17]: Thank you for the comments 😄
Added this to our techincal discussion.

File: internal/service/advancedclustertpf/data_source.go (hunk 1 above)
Comment (EspenAlbert) [reply internal/service/advancedclustertpf/data_source.go]: Thanks. Changed in 13638b93527c2a25df1d139b4e6ab669252985ae

File: internal/service/advancedclustertpf/data_source.go (hunk 2 above)
Comment (EspenAlbert) [reply internal/service/advancedclustertpf/data_source.go]: Good point. Added the link to our migration guide that has more details in 1c35df47ce5467323b0ec5d37c32246baff14536

**Pull request merged**